src/main/webapp/app/exercises/programming/shared/instructions-render/service/programming-exercise-plant-uml.service.ts

```diff
@@ -19,6 +21,10 @@ export class ProgrammingExercisePlantUmlService {
         private themeService: ThemeService,
     ) {
         this.encoder = new HttpUrlCustomEncoder();
+        this.themeService
+            .getCurrentThemeObservable()
+            .pipe(tap(() => themeChangedSubject.next()))
```

Comment:
Isn't it possible to directly subscribe to the ThemeObservable?

Reply:
No, since the service is only present after calling the constructor, but the @Cachable-annotation needs a "static" value. The same approach is used for cache eviction on logout at other places.